### PR TITLE
Fix failing tests in `tests/splinter_test.py`.

### LINKED ISSUE #1624: Fix failing tests in `tests/splinter_test.py`.
##### Fix These Failing Test Cases

- [x] `test_add_asset_image_upload` (`splinter_test.WebTest`)
- [x] `test_add_asset_streaming` (`splinter_test.WebTest`)
- [x] `test_add_asset_url` (`splinter_test.WebTest`)
- [x] `test_add_asset_video_upload` (`splinter_test.WebTest`)
- [x] `test_add_two_assets_upload` (`splinter_test.WebTest`)
- [x] `test_disable_asset` (`splinter_test.WebTest`)
- [x] `test_edit_asset` (`splinter_test.WebTest`)
- [x] `test_enable_asset` (`splinter_test.WebTest`)
- [x] `test_reorder_asset` (`splinter_test.WebTest`)
- [x] `test_rm_asset` (`splinter_test.WebTest`)
- [x] `test_settings_page_should_work` (`splinter_test.WebTest`)
- [x] `test_system_info_page_should_work` (`splinter_test.WebTest`)

##### Additional Tasks

- [x] Resolve *SonarCloud* issues.
- [x] Even though all the test cases are now passing, find a way to remove the following &ndash; "Unable to get local Git branch" & "Unable to get latest version from GitHub"
- [x] Execute `ffmpeg` installation during build time (at a Dockerfile) instead. You can take advantage of multi-stage builds.
- [x] Modify README.md and other relevant docs.

#### Overview of the Issue

```
test_add_asset_image_upload (splinter_test.WebTest) ... ERROR
test_add_asset_streaming (splinter_test.WebTest) ... ERROR
test_add_asset_url (splinter_test.WebTest) ... ERROR
test_add_asset_video_upload (splinter_test.WebTest) ... ERROR
test_add_two_assets_upload (splinter_test.WebTest) ... ERROR
test_disable_asset (splinter_test.WebTest) ... ERROR
test_edit_asset (splinter_test.WebTest) ... ERROR
test_enable_asset (splinter_test.WebTest) ... ERROR
test_reorder_asset (splinter_test.WebTest) ... ERROR
test_rm_asset (splinter_test.WebTest) ... ERROR
test_settings_page_should_work (splinter_test.WebTest) ... ERROR
test_system_info_page_should_work (splinter_test.WebTest) ... ERROR

...

Ran 12 tests in 0.011s

FAILED (errors=12)
```

```
======================================================================
ERROR: test_add_asset_image_upload (splinter_test.WebTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/app/tests/splinter_test.py", line 128, in test_add_asset_image_upload
    with Browser() as browser:
  File "/usr/local/lib/python2.7/dist-packages/splinter/browser.py", line 92, in Browser
    return get_driver(driver, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/splinter/browser.py", line 70, in get_driver
    raise err
SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line
```

##### First Time Setup

```bash
docker-compose -f docker-compose.dev.yml build
docker-compose -f docker-compose.dev.yml up -d
docker-compose -f docker-compose.dev.yml exec srly-ose-test bash bin/prepare_test_environment.sh -s
```

##### Running the Tests

```bash
docker-compose -f docker-compose.dev.yml exec srly-ose-test nosetests -v -a 'fixme' tests/splinter_test.py
```

#### Environment

- WSL 2 (Ubuntu 22.04, x86, 64-bit)
- Docker Desktop for Windows (accessible via WSL 2)

#### Logs (Latest Comes First)

- [test-run-1624-20221011-1103.log](https://github.com/Screenly/screenly-ose/files/9758676/test-run-1624-20221011-1103.log)
- [issue_1624.log](https://github.com/Screenly/screenly-ose/files/9718948/issue_1624.log)

#### Related PRs

- #1635
- ~#1633~

### PR DESCRIPTION
**Pull request by nicomiguelino**
Might fix #1624

### CODE REVIEW

**Comment by nicomiguelino** [comment]
#### Logs (Commit f1602035c18c9b2f6f4df44892f50757ef889e68)

- [test-run-1624-20221011-2028.log](https://github.com/Screenly/screenly-ose/files/9761291/test-run-1624-20221011-2028.log)

#### Overview

```
test_add_asset_url (splinter_test.WebTest) ... ERROR

======================================================================
ERROR: test_add_asset_url (splinter_test.WebTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/app/tests/splinter_test.py", line 71, in test_add_asset_url
    with Browser() as browser:
  File "/usr/local/lib/python2.7/dist-packages/splinter/browser.py", line 92, in Browser
    return get_driver(driver, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/splinter/browser.py", line 70, in get_driver
    raise err
SessionNotCreatedException: Message: Expected browser binary location, but unable to find binary in default location, no 'moz:firefoxOptions.binary' capability provided, and no binary flag set on the command line
```

**Comment by nicomiguelino** [comment]
#### Logs (Commit 42e6f1d45013ce7c8f4d28f2c24f0c1c1420b8d2)

```
test_add_asset_image_upload (splinter_test.WebTest) ... ok
test_add_asset_url (splinter_test.WebTest) ... ok

----------------------------------------------------------------------
Ran 2 tests in 14.692s

OK
```

**Comment by nicomiguelino** [comment]
Hello @vpetersson & @rusko124. I'm not yet done with all of the test cases inside `splinter_test.py` but would like to ask for a review of the changes in files related to CI, Docker, and other scripts.

**Review by vpetersson — APPROVED** [review]
Nice one, @nicomiguelino

File: bin/prepare_test_environment.sh [hunk 1]
```diff
@@ -32,15 +33,26 @@ while getopts "sh" arg; do
...
-cp ansible/roles/screenly/files/screenly.db ~/.screenly/
-cp ansible/roles/screenly/files/screenly.conf ~/.screenly/
-curl $OSE_LOGO_URL > /tmp/image.png
+cp tests/assets/asset.mov /tmp/asset.mov
+cp static/img/ose-logo.png /tmp/image.png
 cp /tmp/image.png /tmp/USB/image.png
 cp /tmp/image.png /tmp/USB/cleanup_folder/image.png
 cp tests/config/ffserver.conf /etc/ffserver.conf
 
+if [ ! -d /data/ffmpeg ] || [ ! -f /data/ffmpeg/ffserver ]; then
```
Comment (vpetersson) [inline bin/prepare_test_environment.sh:42]: Isn't it better to include this in the Dockerfile?

File: docker/Dockerfile.test [hunk 2]
```diff
@@ -1,9 +1,32 @@
-FROM screenly/srly-ose-base:latest-x86
+FROM screenly/srly-ose-base:latest-x86 as builder
```
Comment (vpetersson) [inline docker/Dockerfile.test:1]: So this is fine. You could use the `srly-ose-base:latest-x86` as the builder base, or just `debian:buster` with whatever extras you need to compile.

File: docker/Dockerfile.test [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+
+WORKDIR /opt
+RUN cd /opt
+RUN git clone https://git.ffmpeg.org/ffmpeg.git
+WORKDIR ffmpeg
+RUN git checkout 2ca65fc7b74444edd51d5803a2c1e05a801a6023 && \
+    ./configure --disable-x86asm && make
+
+FROM builder
```
Comment (vpetersson) [inline docker/Dockerfile.test:10]: This is where it goes wrong. You need to use `FROM screenly/srly-ose-base:latest-x86` here. Consider everything *above* this throwaway. You just want to *copy* data from it. You don't want to actually use this image at al.l.

**Review by vpetersson — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.test (hunk 2 above)
Comment (nicomiguelino) [reply docker/Dockerfile.test:1]: Got it, thanks! I'm currently trying out `srly-ose-base:latest-x86` as the base image. Will try using `debian:buster` later.

**Review by vpetersson — APPROVED** [review]
Some minor change requests but LGTM otherwise.

File: docker/Dockerfile.test [hunk 4]
```diff
@@ -1,9 +1,42 @@
+FROM debian:buster as builder
+
+ENV GIT_SSL_NO_VERIFY=1
```
[the remark this answers is not in the document]
Comment (nicomiguelino) [reply docker/Dockerfile.test]: All good, thanks!

[the remark this answers is not in the document]
Comment (nicomiguelino) [reply docker/Dockerfile.test]: You can check out my newer commit.

**Pull request merged**